### ISSUE TITLE
Cilium no longer runs on Bottlerocket Linux as of 1.11.3

### ISSUE THREAD

**Issue by alex-berger** [cilium/cilium#19256]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

With the changes introduced in https://github.com/cilium/cilium/pull/18897 the Cilium agent no longer starts on [Bottlerocket Linux](https://github.com/bottlerocket-os/bottlerocket) nodes.

This is because the node-init DaemonSet cannot write the timestamp to the `bootstrapFile` ("/tmp/cilium-bootstrap-time") in https://github.com/cilium/cilium/blob/v1.11.3/install/kubernetes/cilium/files/nodeinit/startup.bash#L123 as there is no `date` command available on Bottlerocket Linux. Instead, it writes an empty `bootstrapFile`, which in turn causes the test for a non-empty `bootstrapFile` at https://github.com/cilium/cilium/blob/v1.11.3/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L386 to fail.

Due to the above described problem, the agent is stuck forever waiting for  the `bootstrapFile` to become non-empty, which will never happen.

 

### Cilium Version

1.11.3

### Kernel Version

5.10.102

### Kubernetes Version

1.21

### Relevant log output

```shell
Waiting on node-init to run...
Waiting on node-init to run...
Waiting on node-init to run...
Waiting on node-init to run...
Waiting on node-init to run...
...
```

### Anything else?

Looks like this is related to https://github.com/cilium/cilium/issues/15393 and that the same work-around applies, disabling the node-init DaemonSet.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pchaigno** [comment]
cc @aanm

**Comment by aanm** [comment]
@alex-berger what's the output of `ls -la /bin` from a [Bottlerocket Linux](https://github.com/bottlerocket-os/bottlerocket) node?

**Comment by alex-berger** [comment]
@aanm bottlerocket has no tools on the host itself, not even `ls` nor is there a shell. Everything is inside a container on bottlerocket.

**Comment by ubaniabalogun** [comment]
Hey is this now a known issue? I'm experiencing errors install Cilium in a bottlerocket EKS node group as well and would like to know if I should debug harder or stop debugging because its broken. Thanks,

**Comment by DamiaPoquet** [comment]
I'm experiencing the same issue. @aanm is it a confirmed bug already?

**Comment by aanm** [comment]
@ubaniabalogun @DamiaPoquet correct this is a confirmed bug

**Comment by carlosjgp** [comment]
You don't need `node-init` (mount bpf FS) when using BottleRocket just disable it

```
nodeinit:
  enabled: false # false when using bottlerocket
```

**Comment by mqsoh** [comment]
I already have a running cluster and I'd like to switch them over to bottlerocket. I tried deleting the cilium-node-init daemonset but the cilium daemonset still expects a daemonset. Setting nodinit: disable in the cilium-config configmap didn't help. Is there a way to reconfigure Cilium to disable the init containers?

**Comment by christarazi** [comment]
We expect to ship a fix for this in v1.13.2.
